### PR TITLE
:sparkles: Manage communication templates for moderation purposes

### PR DESCRIPTION
**Pull request by foysalit**
To notify users of various moderation actions, warnings etc. we send templated emails with customized content as necessary. This PR makes these templates a module in the system that can be managed by the moderators (i.e: add, update, remove templates). With the newly added xrpc endpoints, moderators can interact with the templates table.

### CODE REVIEW

File: lexicons/com/atproto/admin/createCommunicationTemplate.json
Comment (bnewbold) [inline lexicons/com/atproto/admin/createCommunicationTemplate.json:20]: "placeholders." (pl)

File: lexicons/com/atproto/admin/createCommunicationTemplate.json
Comment (bnewbold) [inline lexicons/com/atproto/admin/createCommunicationTemplate.json:18]: I'm sometimes a fan of indicating when a field has syntax, eg `contentMarkdown`.

File: packages/ozone/src/communication-service/template.ts
Comment (devinivy) [inline packages/ozone/src/communication-service/template.ts:31]: There might be some value in being explicit about which fields are passed into `values()`, and same goes with `set()` in the update method.  I notice that the request body is spread into `template` within the handlers, and it may specify some columns that don't exist or internal columns that we don't want to allow being set.

File: packages/ozone/src/api/admin/updateCommunicationTemplate.ts
Comment (devinivy) [inline packages/ozone/src/api/admin/updateCommunicationTemplate.ts:26]: It's small peas, but it might be nice to maintain the `updatedAt` and `createdAt` times as logic within the template service rather than have any/all callers deal with it.  It's not uncommon for it to be set even lower in the stack, e.g. setting `createdAt` via a default to `now()` within the database itself.

**Review by devinivy — COMMENTED** [review]
This is looking great—very tidy!  Left comments on just a few little details that might merit another peek.

File: packages/ozone/src/db/migrations/20240116T085607200Z-communication-template.ts
Comment (devinivy) [inline packages/ozone/src/db/migrations/20240116T085607200Z-communication-template.ts:4]: Open question: should we seed any default templates in here?

File: lexicons/com/atproto/admin/createCommunicationTemplate.json
Comment (dholms) [reply lexicons/com/atproto/admin/createCommunicationTemplate.json:3]: let's discuss this in ozone sync 👍 

but yes, i'm inclined to move this stuff out of the atproto namespace. whether that's to a new ozone namespace or possibly outside of lexicon in general

but also don't think it needs to hold up this work & we can do a big port over of ozone-specific methods soon in a separate chunk of work

File: packages/ozone/src/communication-service/template.ts
Comment (foysalit) [reply packages/ozone/src/communication-service/template.ts:31]: yeah that makes sense!

File: packages/ozone/src/db/migrations/20240116T085607200Z-communication-template.ts
Comment (foysalit) [reply packages/ozone/src/db/migrations/20240116T085607200Z-communication-template.ts:4]: I think the idea was to not expose any of the templates we use internally. So, we could seed with a sample template just as a starting point (which seems a bit unnecessary) but we won't be seeding with our current templates for sure.

File: packages/ozone/src/api/admin/updateCommunicationTemplate.ts
Comment (foysalit) [reply packages/ozone/src/api/admin/updateCommunicationTemplate.ts:26]: makes sense, I see the default being set at the service layer in other places so going to use that approach.

File: lexicons/com/atproto/admin/updateCommunicationTemplate.json
Comment (foysalit) [reply lexicons/com/atproto/admin/updateCommunicationTemplate.json:12]: hmm.... I don't expect anyone else to be using these apis but making them option means we should throw an error from the handler if these are missing until auth call fill it in?

Comment (devinivy) [reply lexicons/com/atproto/admin/updateCommunicationTemplate.json:12]: > I don't expect anyone else to be using these apis

I think we expect others to run their own services that use these same APIs.  We can coordinate releases for ourselves but it's not as easy to do for others, so I think if it's easy to avoid breaking lexicon changes that would be ideal!

> making them option means we should throw an error from the handler if these are missing until auth call fill it in?

Yeah, I think so!  That gives us the same behavior for ourselves but is more flexible at the lexicon level, where we know/expect we'll make use of it.

File: packages/ozone/tests/communication-templates.test.ts
Comment (devinivy) [reply packages/ozone/tests/communication-templates.test.ts]: Yeah using the `nullable` setting in lexicon seems like a good call for that 👍 Sounds good, just wanted to ensure we have a plan for how patches should work since we don't really have any other examples of it anywhere.

Comment (foysalit) [reply packages/ozone/tests/communication-templates.test.ts]: but since there are no fields that we can "unset" here, I feel like it's fine as is? I guess one could try to unset subject but there's no use case for that and could be done by passing `subject: ''`.

Comment (devinivy) [reply packages/ozone/tests/communication-templates.test.ts]: I was just seeing what the plan would be if we want to support unsetting a field w/ patches—if there are none to unset right now then that's fine!

File: lexicons/com/atproto/admin/createCommunicationTemplate.json
Comment (bnewbold) [reply lexicons/com/atproto/admin/createCommunicationTemplate.json:3]: decision/resolution from the sync meeting is to get a new domain name to use as an NSID namespace, and move ozone-specific stuff (like these email template endpoints) there. but we can potentially do that as a bigger cleanup after this PR lands.

File: lexicons/com/atproto/admin/defs.json
Comment (bnewbold) [reply lexicons/com/atproto/admin/defs.json:600]: yeah, leaving door open to UUID or similar in alt implementations would be nice

**Review by bnewbold — APPROVED** [review]
some small comments on lexicon, overall looks good!

**Comment by foysalit** [comment]
Alright I think I've addressed everything. LMK when this can be merged and deployed so that I can make necessary adjustments to the ozone repo. @devinivy / @bnewbold

**Comment by bnewbold** [comment]
looks good to me, but I would wait for divy/dholms approval and deploy guidance

**Pull request merged**
